community/front-end/ofe/website/ghpcfe/views/images.py

```diff
@@ -0,0 +1,290 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" clusters.py """
+
+import os
+from asgiref.sync import sync_to_async
+from django.shortcuts import get_object_or_404
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.views import redirect_to_login
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.core.exceptions import PermissionDenied
+from django.urls import reverse_lazy
+from django.conf import settings
+from django.http import (
+    HttpResponseRedirect,
+    JsonResponse,
+)
+from django.urls import reverse
+from django.views import generic
+from django.views.generic.edit import CreateView
+from ..models import StartupScript, Image, Credential
+from ..forms import StartupScriptForm, ImageForm
+from ..cluster_manager.image import ImageBackend
+from ..cluster_manager.cloud_info import get_region_zone_info
+from ..views.asyncview import BackendAsyncView
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ImagesListView(LoginRequiredMixin, generic.ListView):
+    """Custom ListView for StartupScript and Images model"""
+
+    model = StartupScript
+    template_name = "image/list.html"
+
+    def get_queryset(self):
+        # If user is admin, return all objects.
+        if self.request.user.has_admin_role():
+            startup_scripts = StartupScript.objects.all()
+            images = Image.objects.all()
+            return startup_scripts, images
+        else:
+            # Retrieve startup scripts and images owned by the user
+            startup_scripts = StartupScript.objects.filter(owner=self.request.user)
+            images = Image.objects.filter(owner=self.request.user)
+            
+            # Retrieve startup scripts and images authorized for the user
+            authorized_startup_scripts = StartupScript.objects.filter(authorised_users=self.request.user)
+            authorized_images = Image.objects.filter(authorised_users=self.request.user)
+            
+            # Combine the owned and authorized objects
+            startup_scripts = startup_scripts | authorized_startup_scripts
+            images = images | authorized_images
+            
+            return startup_scripts, images
+        
+    def get_context_data(self, *args, **kwargs):
+        loading = 0
+        admin_view = 0        
+        if self.request.user.has_admin_role():
+            admin_view = 1
+        context = super().get_context_data(*args, **kwargs)
+        context["loading"] = loading
+        context["admin_view"] = admin_view
+        context["navtab"] = "image"
+
+        startup_scripts, images = self.get_queryset()
```

Comment:
Could be written as `context["startupscripts"], context["images"] = self.get_queryset()`, although that may be difficult to understand